src/components/exam/question-types/fill-in-blank.tsx

```
@@ -55,47 +255,117 @@ export function FillInBlankQuestion({ question, onAnswerChange }: FillInBlankQue
-            {/* Question content with blanks indicated */}
-            <div className="prose prose-sm max-w-none dark:prose-invert">
-                <ContentPreview content={(question.question as string) || ""} />
-            </div>
+                {hasInlineBlanks ? (
+                    /* ── Inline mode: inputs appear in-line within the question text ── */
+                    <div className="leading-loose text-sm text-foreground">
+                        {questionSegments.map((segment, i) => (
+                            <React.Fragment key={i}>
+                                <span
+                                    className="prose prose-sm dark:prose-invert *:inline"
+                                    dangerouslySetInnerHTML={{ __html: segment }}
```

Biome (2.4.4)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

(lint/security/noDangerouslySetInnerHtml)